packages/server/modules/core/services/streams.js

```diff
@@ -1,13 +1,85 @@
 'use strict'
+const _ = require('lodash')
 const crs = require('crypto-random-string')
-const appRoot = require('app-root-path')
-const knex = require(`${appRoot}/db/knex`)
+const debug = require('debug')
 
-const Streams = () => knex('streams')
-const Acl = () => knex('stream_acl')
+const { createBranch } = require('@/modules/core/services/branches')
+const { Streams, StreamAcl, knex, StreamFavorites } = require('@/modules/core/dbSchema')
+
+/**
+ * List of columns to select when querying for user streams
+ * (expects joins to StreamAcl)
+ */
+const userStreamColumns = [
+  Streams.col.id,
+  Streams.col.name,
+  Streams.col.description,
+  Streams.col.isPublic,
+  Streams.col.createdAt,
+  Streams.col.updatedAt,
+  StreamAcl.col.role
+]
 
-const debug = require('debug')
-const { createBranch } = require('./branches')
+/**
+ * Get base query for finding or counting user streams
+ * @param {object} p
+ * @param {string} [p.searchQuery] Filter by name/description/id
+ * @param {boolean} [p.publicOnly] Whether to only look for public streams
+ * @param {string} p.userId The user's ID
+ */
+function getUserStreamsQueryBase({ userId, publicOnly, searchQuery }) {
```

Comment:
Moved out this setup to its own function cause it was duplicated in `getUserStreams` and `getUserStreamsCount`